lib/handler/configurator/busypoll.c

```diff
@@ -0,0 +1,139 @@
+#include "h2o.h"
+#include "h2o/configurator.h"
+
+#include <net/if.h>
+
+static int on_busy_poll_map(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
```

Comment:
We seem to have config knobs defined in this file and also in src/main.c. Could you elaborate what the distinction between the two are? (FWIW I would not mind everything going into src/main.c, though I generally share the concern that the file is becoming too large)

Reply:
I've done some refactoring to minimize loop changes and avoid impacting non-linux platforms. In hindsight it might have been worthwhile making busypoll.h cross platform with a few guards around the cpu set.

Reply:
Thank you for the changes! I see the build succeeding on macOS and linux without ynl.

> In hindsight it might have been worthwhile making busypoll.h cross platform with a few guards around the cpu set.

Yeah I think my comments regarding the direction is based on the assumption that support for busy poll will be an option to the evloop-epoll backend based on if libynl is available, when we have 4 backends (libuv, evloop-epoll, evloop-kqueue, evloop-poll). Also, the busy poll code might change.

Considering these aspects, I have been inclined to not have an abstraction layer, but to keep the code specific as possible.